In [13]:
import numpy as np
import pandas as pd

In [14]:
df = pd.read_csv('cleaned_numeric_data.csv')
df.head()

TotalBsmtSF  YearBuilt  YearRemodAdd  LotArea  OverallCond  SalePrice
0        856.0       2003          2003     8450            5   208500.0
1       1262.0       1976          1976     9600            8   181500.0
2        920.0       2001          2002    11250            5   223500.0
3        756.0       1915          1970     9550            5   140000.0
4       1145.0       2000          2000    14260            5   250000.0

In [15]:
df.shape

(1460, 6)

In [16]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1460 entries, 0 to 1459
Data columns (total 6 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   TotalBsmtSF   1460 non-null   float64
 1   YearBuilt     1460 non-null   int64  
 2   YearRemodAdd  1460 non-null   int64  
 3   LotArea       1460 non-null   int64  
 4   OverallCond   1460 non-null   int64  
 5   SalePrice     1460 non-null   float64
dtypes: float64(2), int64(4)
memory usage: 68.6 KB


In [17]:
X = df.drop(columns = ['SalePrice'])
y = df['SalePrice']

In [18]:
from sklearn.model_selection import train_test_split

X_train , X_test , y_train , y_test = train_test_split(
    X , y , test_size = 0.2 , random_state = 42
)

Date is split in 80% training and 20% testing ste

In [19]:
from sklearn.linear_model import LinearRegression

lr = LinearRegression()
lr.fit(X_train , y_train)
y_pred = lr.predict(X_test)

Linear Regression model trained using sklearn

In [20]:
lr.coef_ , lr.intercept_

(array([7.12251320e+01, 6.08888628e+02, 9.05512448e+02, 1.09126100e+00,
        5.19838336e+03]),
 np.float64(-2932327.9930071468))

In [21]:
from sklearn.metrics import mean_squared_error, r2_score

rmse = np.sqrt(mean_squared_error(y_test , y_pred))
r2 = r2_score(y_test , y_pred)

print('RMSE :' , rmse)
print('R2 Score :' , r2)

RMSE : 57388.70873622514
R2 Score : 0.5706227086762887


In [22]:
coeff_df = pd.DataFrame({
    'Feature': X.columns,
    'Coefficient': lr.coef_
})

coeff_df.sort_values(by='Coefficient', ascending=False)

Feature  Coefficient
4   OverallCond  5198.383359
2  YearRemodAdd   905.512448
1     YearBuilt   608.888628
0   TotalBsmtSF    71.225132
3       LotArea     1.091261

## Model Evaluation

- **Train-Test Split:** 80–20
- **RMSE:** 57,388.71
- **R² Score:** 0.57

### Observation
The model explains about 57% of the variance in house prices.  
This is a baseline Linear Regression model, and performance is limited due to
linear assumptions and use of numeric features only.
